Library/Homebrew/download_strategy.rb

```diff
@@ -329,7 +329,9 @@ def resolved_url_and_basename
   end
 
   def parse_basename(url, search_query: true)
-    uri_path = if url.match?(URI::DEFAULT_PARSER.make_regexp)
+    components = { path: T.let([], T::Array[String]), query: T.let([], T::Array[String]) }
```

Comment:
Are all the `T.let` necessary? What happens without them?

Reply:
```
download_strategy.rb:346: Expected [] but found T::Array[T.untyped] for key Symbol(:path) https://srb.help/7002
     346 |        components[:path] = path.split("/").map do |part|
     347 |          URI::DEFAULT_PARSER.unescape(part).presence
     348 |        end.compact
  Shape originates from here:
    download_strategy.rb:332:
     332 |    components = { path: [], query: [] }
                           ^^^^^^^^^^^^^^^^^^^^^^^
  Got T::Array[T.untyped] originating from:
    download_strategy.rb:346:
     346 |        components[:path] = path.split("/").map do |part|
     347 |          URI::DEFAULT_PARSER.unescape(part).presence
     348 |        end.compact

download_strategy.rb:352: Expected [] but found T::Array[T.untyped] for key Symbol(:query) https://srb.help/7002
     352 |        components[:query] = URI.decode_www_form(query).map(&:second)
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Shape originates from here:
    download_strategy.rb:332:
     332 |    components = { path: [], query: [] }
                           ^^^^^^^^^^^^^^^^^^^^^^^
  Got T::Array[T.untyped] originating from:
    download_strategy.rb:352:
     352 |        components[:query] = URI.decode_www_form(query).map(&:second)
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Errors: 2
```

Reply:
Cool, ignore me then!